ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/optimisation/fillers/MaxLoopFlowFiller.java

```diff
@@ -57,29 +73,70 @@ private void buildMaxLoopFlowConstraint(LinearRaoData linearRaoData, LinearRaoPr
+        double loopflowViolationCost = 0.0;
+        if (!Objects.isNull(linearRaoParameters.getExtendable())) {
+            loopflowConstraintAdjustmentCoefficient = linearRaoParameters.getExtendable().getLoopflowConstraintAdjustmentCoefficient();
+            loopflowViolationCost = linearRaoParameters.getExtendable().getLoopflowViolationCost();
+        }
+
         for (Cnec cnec : linearRaoData.getCrac().getCnecs(linearRaoData.getCrac().getPreventiveState())) {
+            //security check
             if (Objects.isNull(cnec.getExtension(CnecLoopFlowExtension.class))) {
                 continue;
             }
-            double loopFlowShift = 0.0;
+
+            //get and update MapLoopflowLimit with loopflowConstraintAdjustmentCoefficient
             double maxLoopFlowLimit = Math.abs(cnec.getExtension(CnecLoopFlowExtension.class).getLoopFlowConstraint());
-            double loopflowConstraintAdjustmentCoefficient = 0.0;
-            if (!Objects.isNull(linearRaoParameters.getExtendable())) {
-                loopflowConstraintAdjustmentCoefficient = linearRaoParameters.getExtendable().getLoopflowConstraintAdjustmentCoefficient();
-            }
             maxLoopFlowLimit = Math.max(0.0, maxLoopFlowLimit - loopflowConstraintAdjustmentCoefficient);
-            if (loopFlowShifts.containsKey(cnec)) {
-                loopFlowShift = loopFlowShifts.get(cnec);
-            }
-            MPConstraint maxLoopflowConstraint = linearRaoProblem.addMaxLoopFlowConstraint(
-                    -maxLoopFlowLimit + loopFlowShift,
-                    maxLoopFlowLimit + loopFlowShift,
-                    cnec);
+
+            //get loopflow shift
+            double loopFlowShift = loopFlowShifts.getOrDefault(cnec, 0.0);
+
+            //get MPVariable: flowVariable
             MPVariable flowVariable = linearRaoProblem.getFlowVariable(cnec);
             if (Objects.isNull(flowVariable)) {
                 throw new FaraoException(String.format("Flow variable on %s has not been defined yet.", cnec.getId()));
             }
-            maxLoopflowConstraint.setCoefficient(flowVariable, 1);
-        }
+
+            if (loopflowViolationCost == 0.0) {
+                //no loopflow violation cost => 1 MP constraint
+                //NOTE: The "zero-loopflowViolationCost" routine handles infeasible / non-optimal solver status in LinearRao.
+                MPConstraint maxLoopflowConstraint = linearRaoProblem.addMaxLoopFlowConstraint(
+                        -maxLoopFlowLimit + loopFlowShift,
+                        maxLoopFlowLimit + loopFlowShift,
+                        cnec);
+                maxLoopflowConstraint.setCoefficient(flowVariable, 1);
+            } else {
+                //loopflow violation cost is not zero => 2 MP constraints, additional MPVariable loopflowBreachVariable
+                MPVariable loopflowBreachVariable = linearRaoProblem.addLoopflowBreachVariable(
```

Comment:
I think we must choose between the term violation and the term breach. I would replace loopflowBreachVariable by loopflowViolationVariable